velox/exec/tests/SharedArbitratorTest.cpp

```diff
@@ -382,6 +383,242 @@ class SharedArbitrationTest : public exec::test::HiveConnectorTestBase {
+    }
+    return task;
+  }
+
+  std::shared_ptr<Task> runOrderByTask(
+      const std::vector<RowVectorPtr>& vectors,
+      const std::shared_ptr<core::QueryCtx>& queryCtx,
+      uint32_t numDrivers) {
+    auto planNodeIdGenerator = std::make_shared<core::PlanNodeIdGenerator>();
+    const auto spillDirectory = exec::test::TempDirectoryPath::create();
+
+    std::shared_ptr<Task> task;
+    try {
+      task = AssertQueryBuilder(duckDbQueryRunner_)
+                 .spillDirectory(spillDirectory->path)
+                 .config(core::QueryConfig::kSpillEnabled, "true")
+                 .config(core::QueryConfig::kOrderBySpillEnabled, "true")
+                 .queryCtx(queryCtx)
+                 .maxDrivers(numDrivers)
+                 .plan(PlanBuilder(planNodeIdGenerator)
+                           .values(vectors)
+                           .project({"c0", "c1", "c2"})
+                           .orderBy(
+                               {fmt::format("{} ASC NULLS LAST", "c2")}, false)
+                           .planNode())
+                 .assertResults(
+                     "SELECT c0, c1, c2 "
+                     "FROM tmp "
+                     "ORDER BY c2 ASC NULLS LAST");
+    } catch (const VeloxException& e) {
+      VELOX_CHECK(
+          e.errorCode() == error_code::kMemCapExceeded.c_str() ||
+          e.errorCode() == error_code::kMemAborted.c_str() ||
+          e.errorCode() == error_code::kMemAllocError.c_str());
+    }
+
+    return task;
+  }
+
+  std::shared_ptr<Task> runRowNumberTask(
+      const std::vector<RowVectorPtr>& vectors,
+      const std::shared_ptr<core::QueryCtx>& queryCtx,
+      uint32_t numDrivers) {
+    auto planNodeIdGenerator = std::make_shared<core::PlanNodeIdGenerator>();
+    const auto spillDirectory = exec::test::TempDirectoryPath::create();
+
+    std::shared_ptr<Task> task;
+    try {
+      task =
+          AssertQueryBuilder(duckDbQueryRunner_)
+              .spillDirectory(spillDirectory->path)
+              .config(core::QueryConfig::kSpillEnabled, "true")
+              .config(core::QueryConfig::kRowNumberSpillEnabled, "true")
+              .queryCtx(queryCtx)
+              .maxDrivers(numDrivers)
+              .plan(PlanBuilder(planNodeIdGenerator)
+                        .values(vectors)
+                        .rowNumber({"c0"}, 2, false)
+                        .project({"c0", "c1"})
+                        .planNode())
+              .assertResults(
+                  "SELECT c0, c1 "
+                  "FROM ("
+                  " SELECT *, row_number() over (partition by c0) AS rn FROM tmp) "
+                  "WHERE rn <= 2");
+    } catch (const VeloxException& e) {
+      VELOX_CHECK(
+          e.errorCode() == error_code::kMemCapExceeded.c_str() ||
+          e.errorCode() == error_code::kMemAborted.c_str() ||
+          e.errorCode() == error_code::kMemAllocError.c_str());
+    }
+
+    return task;
+  }
+
+  std::shared_ptr<Task> runTopNTask(
```

Comment:
Shall we move the exception catch in the caller instead of inside runXXXTask function? Thanks!

Reply:
Fixed